### PR TITLE
Opens wallet from Ethereum compatible private key

### PR DESCRIPTION
**Pull request by ysz**

### CODE REVIEW

File: src/app/state/wallet/types.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export enum WalletType {
   Ledger = 'ledger',
   PrivateKey = 'private_key',
   Mnemonic = 'mnemonic',
+  ParaTime = 'paratime',
```
Comment (lukaw3d) [inline src/app/state/wallet/types.ts:5]: This is ethereum-specific, not paratime

File: src/app/state/account/saga.ts [hunk 2]
```diff
@@ -25,12 +30,44 @@ function* loadAccount(action: PayloadAction<string>) {
...
+            const newAddress = yield* call(oasis.staking.addressFromBech32, paratimeAddress)
+
+            const balancesResult = yield* call((nicArg) => accountsWrapper
+              .queryBalances()
+              .setArgs({
+                address: newAddress,
+              })
+              .query(nic)
+              .catch((err: any) => {
+                return err
```
Comment (lukaw3d) [inline src/app/state/account/saga.ts:47]: no.

File: src/app/state/wallet/saga.ts [hunk 3]
```diff
@@ -122,6 +123,22 @@ export function* openWalletFromMnemonic({ payload: mnemonic }: PayloadAction<str
   )
 }
 
+export function* openWalletFromEthereumPrivateKey({ payload: tmp }: PayloadAction<any>) {
```
Comment (lukaw3d) [inline src/app/state/wallet/saga.ts:126]: bad name and type

File: src/app/pages/AccountPage/index.tsx [hunk 4]
```diff
@@ -91,7 +92,7 @@ export function AccountPage(props: Props) {
 
   // Reload account balances if address or network changes
   useEffect(() => {
-    dispatch(accountActions.fetchAccount(address))
+    dispatch(accountActions.fetchAccount({address,type:walletType}))
```
Comment (lukaw3d) [inline src/app/pages/AccountPage/index.tsx:95]: This is wrong: address comes from URL

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    const key_bytes = base64ToUint(keyWithoutEnvelope)
+    return OasisKey.fromPrivateKey(key_bytes)
+  } catch (e) {
+    throw e
+  }
+}
+
+export async function getEvmBech32Address(evmAddress: any) {
+  if (!evmAddress) {
+    return ''
```
Comment (lukaw3d) [inline src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx:32]: no.

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  }
+  let newEvmAddress = evmAddress
+  if (newEvmAddress.indexOf('0x') === 0) {
+    newEvmAddress = newEvmAddress.substr(2)
+  }
+  const evmBytes = oasis.misc.fromHex(newEvmAddress)
```
Comment (lukaw3d) [inline src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx:38]: `oasis.misc.fromHex(evmAddress.replace('0x', ''))`

This also gets rid of the odd name "newEvmAddress"

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+      .replace(/-----.*?-----$/, '')
+
+    const key_bytes = base64ToUint(keyWithoutEnvelope)
+    return OasisKey.fromPrivateKey(key_bytes)
+  } catch (e) {
+    throw e
+  }
+}
+
+export async function getEvmBech32Address(evmAddress: any) {
```
Comment (lukaw3d) [inline src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx:30]: Type

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  const { t } = useTranslation()
+  const walletActions = useWalletSlice().actions
+  const dispatch = useDispatch()
+
+  const [privateKey, setPrivateKey] = React.useState('')
+  const [privateKeyIsValid, setPrivateKeyIsValid] = React.useState(true)
+  const [privateKeyIsVisible, setPrivateKeyIsVisible] = React.useState(false)
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => setPrivateKey(event.target.value)
+  const onSubmit = () => {
```
Comment (lukaw3d) [inline src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx:57]: Most of this should be in saga

Comment (ysz) [reply src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx:21]: This has been refactored.

File: src/app/pages/AccountPage/index.tsx (hunk 4 above)
Comment (ysz) [reply src/app/pages/AccountPage/index.tsx:95]: How to reproduce: Use Search box and search for address.

File: src/app/state/wallet/types.ts (hunk 1 above)
Comment (buberdds) [reply src/app/state/wallet/types.ts]: done

File: src/app/state/wallet/saga.ts (hunk 3 above)
Comment (buberdds) [reply src/app/state/wallet/saga.ts]: done

File: src/app/pages/AccountPage/index.tsx (hunk 4 above)
Comment (buberdds) [reply src/app/pages/AccountPage/index.tsx]: done

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx (hunk 6 above)
Comment (buberdds) [reply src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx]: FromEthereumPrivateKey has only view logic now

File: src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx (hunk 7 above)
Comment (buberdds) [reply src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx]: done

**Pull request closed without merging**